### ISSUE TITLE
Sketchup :: Page#use_rendering_options_ vs  #use_style_ docs suggestion

### ISSUE THREAD

**Issue by dezmoHU** [SketchUp/api-issue-tracker#676]
[#**use_rendering_options=** (setting) ⇒ Object](https://ruby.sketchup.com/Sketchup/Page.html#use_rendering_options=-instance_method)
[#**use_rendering_options?** ⇒ Boolean](https://ruby.sketchup.com/Sketchup/Page.html#use_rendering_options%3F-instance_method)

[#**use_style=** (style) ⇒ Object](https://ruby.sketchup.com/Sketchup/Page.html#use_style=-instance_method)
[#**use_style?** ⇒ Boolean](https://ruby.sketchup.com/Sketchup/Page.html#use_style%3F-instance_method)

Perhaps could be good to mention in the documentation:
If you are setting the `#use_rendering_optoins=` it will also toggle the page to use the style. 
And and vice versa: 
If you are applying a style to the page  it will force the rendering options to use.

*It is interesting to see that all other setters use a `Boolean` (TrueClass or FalseClass) to set, but the `#use_style=` is using the style object.*

console:
```
page.use_rendering_options?
true
page.use_style?
true
style = page.style
#<Sketchup::Style:0x000002685c6b9010>
page.use_rendering_options = false
false
page.use_style?
false
page.use_style = style
#<Sketchup::Style:0x000002685c6b9010>
page.use_style?
true
page.use_rendering_options?
true
```
See details:
[SU Forum](https://forums.sketchup.com/t/thoughts-on-class-sketchup-page-mostly-docs-issues/169968?u=dezmo)

**Comment by DanRathbun** [comment]
> It is interesting to see that all other setters use a `Boolean` (TrueClass or FalseClass) to set, but the `#use_style=` is using the style object.

Yes I agree. Other places in the API the setter methods that take an object or a set of parameters begin with the word "set" as in "set_style=" or simply "style=". (One example would be the [`Layout::Entity#style=`](https://ruby.sketchup.com/Layout/Entity.html#style=-instance_method) setter.)

> If you are setting the `#use_rendering_optoins=` it will also toggle the page to use the style.
> And and vice versa: If you are applying a style to the page it will force the rendering options to use.

This is true because the rendering options interface predated the style interface. After style was implemented, the rendering options object remained for backward compatibility. 

Also there is **not yet** direct manipulation with a SketchUp style object via the API. (All you can do, other than choose an active style, is load a style and edit it's name or description.) To make changes to a loaded style, coders must manipulate a rendering options object, select one of the model's styles ([`Styles#selected_style=`](https://ruby.sketchup.com/Sketchup/Styles.html#selected_style=-instance_method)) and then update it ([`Styles#update_selected_style`](https://ruby.sketchup.com/Sketchup/Styles.html#update_selected_style-instance_method)) with the current rendering options settings. (The code example of the latter update method does little to inform readers of the dependence of the two classes upon each other.)

So upon understanding the history, and the interaction between the two classes, it should make sense that the two boolean flags are linked.

It would probably benefit coders to know these facts. This information could be added to the [`Style`](https://ruby.sketchup.com/Sketchup/Style.html) class overview. 

The [`RenderingOptions`](https://ruby.sketchup.com/Sketchup/RenderingOptions.html) class overview has only this vague statement:
"_The majority of the rendering information returned exists in the Styles dialog_".
A note with a link to the [`Style`](https://ruby.sketchup.com/Sketchup/Style.html) class might be helpful.

**Comment by thomthom** [comment]
> It is interesting to see that all other setters use a Boolean (TrueClass or FalseClass) to set, but the #use_style= is using the style object.

Ruby apply some special magic for any methods that ends with `=`; it will override any `return` value in the method and always return the input parameter.

**Comment by DanRathbun** [comment]
@Eneroth3 I think this is working as intended. (See my comments above.) I also think I brought this up many years ago and Chris said it was working as intended.

**Comment by Eneroth3** [comment]
I agree this is working as intended. `use_rendering_options=` corersponds to the checkbox "Styles and Fog" in the UI while `use_style=` corresponds to selecting a different style and updating the style for the scene.

The documentation can be clearer here. (And ideally `use_style=` should just have been called `style=`)

**Comment by DanRathbun** [comment]
> while `use_style=` corresponds to selecting a different style and updating the style for the scene.

It will **also** do the update ? (The [doc ](https://ruby.sketchup.com/Sketchup/Page.html#use_style=-instance_method)does not say.)
